### ISSUE TITLE
Documentation inconsistency

### ISSUE THREAD

**Issue by osadi** [mikaelvesavuori/mikrovalid#3]
In [README](https://github.com/mikaelvesavuori/mikrovalid/blob/main/README.md?plain=1#L102) at line 102 the example says:

```json
{
  "properties": {
    "username": {
      "type": "string"
    }
  },
  "required": ["name"]
}
```
Where the required property `name` does not exist in the properties declaration, or the valid input in the following paragraph.

Would a PR with a suggestion on stricter types for the schema input be of interest?

----

In [MikroValid.ts](https://github.com/mikaelvesavuori/mikrovalid/blob/main/src/domain/MikroValid.ts#L35), the example has the `required` properties on the same level as the actual properties.

In the tests and in the README, those properties are on the same level as the `properties` property.

The way they are placed in the code description is similar to how they are placed in the rest of the hierarchy, like
```typescript
work: {
  office: 'string', // <- referenced by required on the same level
  currency: 'string',
  salary: 'number',
  required: ['office'] // <- here
},
// ---------- Tests and README
{
  properties: {
    thing: {},  // <- referenced by required on one level up
  },
  required: ['thing'] // <- only on this level for things inside properties?
}
```

If this is an inconsistency, and it would be addressed, it is also valid for `additionalProperties` . (Not explicitly written in the code description though).

----
Also in the code description:

```typescript
work: {
  office: 'string', // <- These should be `{ type: 'string' }` ?
  currency: 'string',
  salary: 'number',
  required: ['office']
},
```

**Comment by mikaelvesavuori** [comment]
Hi again @osadi!

All of this is due to human error and sloppiness. I've fixed this in `1.0.8` (see https://github.com/mikaelvesavuori/mikrovalid/releases/tag/v1.0.8). Thanks for reporting!

`Would a PR with a suggestion on stricter types for the schema input be of interest?` --> Sure!

**Comment by mikaelvesavuori** [comment]
Just adding that `additionalProperties` can be used at any level inside the initial properties object - otherwise you could not describe that condition for each nested level.
